static/js/stream_data.js

```diff
@@ -1,6 +1,66 @@
 const Dict = require('./dict').Dict;
 const LazySet = require('./lazy_set').LazySet;
 
+const BinaryDict = function (pred) {
```

Comment:
This would probably benefit from a docstring type comment, just so one doesn't have to go back to the original commit to understand what this is for.